tests/integration/integration-test.py

```diff
@@ -34,73 +38,141 @@ def setup(cls, config_path, openapi_path):
-                                           expected_status_code,
-                                           query_params=params,
-                                           nullable_fields=nullable_fields)
-            content = utils.get_json_content(self, response)
-            if expected_status_code == 200:
-                try:
-                    meta = content['meta']
-                    num = pagination['number'] if pagination['number'] else 1
-                    size = pagination['size'] if pagination['size'] else 25
-
-                    self.assertEqual(num, meta['currentPageNumber'])
-                    self.assertEqual(size, meta['currentPageSize'])
-                except KeyError as error:
-                    self.fail(error)
-
-    # Test case: GET /pets/{id}
-    def test_get_pet_by_id(self, endpoint='/pets'):
-        valid_pet_ids = self.test_cases['valid_pet_ids']
-        invalid_pet_ids = self.test_cases['invalid_pet_ids']
-
-        for pet_id in valid_pet_ids:
-            resource = 'PetResource'
-            utils.test_endpoint(self, f'{endpoint}/{pet_id}', resource, 200)
-
-        for pet_id in invalid_pet_ids:
-            resource = 'ErrorObject'
-            utils.test_endpoint(self, f'{endpoint}/{pet_id}', resource, 404)
+                actual_state = resource['attributes']['location']['state']
+                self.assertEqual(actual_state, state)
+        for state in invalid_states:
+            params = {'filter[state]': state}
+            utils.test_endpoint(self, endpoint, 'ErrorObject', 400,
+                                query_params=params)
+
+    # Test GET parks/{id}
+    def test_get_park_by_id(self):
+        valid_park_ids = self.test_cases['valid_park_ids']
+        invalid_park_ids = self.test_cases['invalid_park_ids']
+        for park_id in valid_park_ids:
+            response = utils.test_endpoint(self, f'/parks/{park_id}', park_res,
+                                           200)
+            actual_id = response.json()['data']['id']
+            self.assertEqual(actual_id, park_id)
+        for parkId in invalid_park_ids:
+            utils.test_endpoint(self, f'/parks/{parkId}', err_obj, 404)
+
+    # Test GET owners/{ownerId}/parks
+    def test_get_park_by_owner_id(self):
+        valid_owner_ids = self.test_cases['valid_owner_ids']
+        invalid_owner_ids = self.test_cases['invalid_owner_ids']
+        for owner_id in valid_owner_ids:
+            response = utils.test_endpoint(self, f'/owners/{owner_id}/parks',
+                                           'ParkResource', 200)
+            response_data = response.json()['data']
+            for park in response_data:
+                actual_id = park['relationships']['owner']['data']['id']
+                self.assertEqual(actual_id, owner_id)
+        for owner_id in invalid_owner_ids:
+            utils.test_endpoint(self, f'/owners/{owner_id}/parks', err_obj,
+                                404)
```

Comment:
This should be tested in `test_get_owner_by_id` (and you actually did, so this part is not actually needed.).